### PR TITLE
Localized assets

### PR DESCRIPTION
**Pull request by vkalpias**
Adds support for localized assets. This PR allows Text Elements to use a different font asset based on the used locale.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/asset/asset.js [hunk 1]
```diff
@@ -122,6 +126,22 @@ Object.assign(pc, function () {
      * @param {*} oldValue The old property value
      */
 
+    /**
+     * @event
+     * @name pc.Asset#set:localized
```
Comment (daredevildave) [inline src/asset/asset.js]: Should this be "add:localized"? To pair add/remove

File: src/asset/asset.js [hunk 2]
```diff
@@ -189,6 +209,49 @@ Object.assign(pc, function () {
...
+         */
+        setLocalizedAssetId: function (locale, assetId) {
+            this._i18n[locale] = assetId;
+            this.fire('set:localized', locale, assetId);
+        },
+
+        /**
+         * @private
+         * @function
+         * @name pc.Asset#setLocalizedAssetId
```
Comment (daredevildave) [inline src/asset/asset.js]: should be remove

File: src/asset/asset.js [hunk 3]
```diff
@@ -189,6 +209,49 @@ Object.assign(pc, function () {
...
+         * @param {String} locale The locale e.g. ar-AR.
+         * @param {Number} assetId The asset id
+         * @description Sets a replacement asset id for the specified locale. When the locale in {@link pc.Application#i18n} changes then
+         * references to this asset will be replaced with the specified asset id. (Currently only supported by the {@link pc.ElementComponent}).
+         */
+        setLocalizedAssetId: function (locale, assetId) {
```
Comment (daredevildave) [inline src/asset/asset.js]: Same comment as the event. Should we pair add/remove rather than set/remove

File: src/asset/asset-localized.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+        this._defaultAsset = null;
+        this._localizedAsset = null;
+
+        pc.events.attach(this);
+    };
+
+    LocalizedAsset.prototype._bindDefaultAsset = function () {
+        var asset = this._app.assets.get(this._defaultAsset);
+        if (!asset) return;
```
Comment (daredevildave) [inline src/asset/asset-localized.js]: Looks like we no longer handle the case where the asset is added to the registry after the element is created?

File: src/framework/components/element/text-element.js [hunk 5]
```diff
@@ -28,7 +28,12 @@ Object.assign(pc, function () {
         this._text = "";
         this._i18nKey = null;
 
-        this._fontAsset = null;
+        this._fontAsset = new pc.LocalizedAsset(this._system.app);
+        this._fontAsset.disableLocalization = true;
+        this._fontAsset.on('load', this._onFontLoad, this);
```
Comment (daredevildave) [inline src/framework/components/element/text-element.js]: I don't see where we fire these events for the default asset?

Comment (vkalpias) [reply src/framework/components/element/text-element.js]: These events are fired on the pc.LocalizedAsset instance

File: src/asset/asset.js (hunk 3 above)
Comment (vkalpias) [reply src/asset/asset.js]: We could do that sure.

File: src/asset/asset-localized.js (hunk 4 above)
Comment (vkalpias) [reply src/asset/asset-localized.js]: We handle it in onSetLocale but we do not handle adding the defaultAsset later when it comes to the set:localized and remove:localized events which I'll fix. Basically whatever ends up in localizedAsset is the asset where we listen for load events etc.

**Pull request merged**